common/djangoapps/enrollment/data.py

```diff
@@ -337,3 +338,15 @@ def get_course_enrollment_info(course_id, include_expired=False):
         raise CourseNotFoundError(msg)
     else:
         return CourseSerializer(course, include_expired=include_expired).data
+
+
+def get_user_roles(user_id):
+    """
+    Returns a list of all roles that this user has.
+    :param user_id: The id of the selected user.
+    :return: All roles for all courses that this user has.
+    """
+    user = _get_user(user_id)
+    if not hasattr(user, '_roles'):
+        user._roles = RoleCache(user)
```

Comment:
Can we have a different name for the cache object? like `user._role_cache`?

Reply:
I didn't come up with these names, they're existing property names

Reply:
I see. Then ignore my feedback. But, don't you agree the naming is confusing?

Reply:
Yes, I definitely agree

Reply:
You could always do:
```python
role_cache = user._roles
return role_cache._roles
```

Reply:
(Instead of `return user._roles._roles`)